crates/matrix-sdk/src/error.rs

```diff
@@ -46,6 +43,16 @@ pub type Result<T, E = Error> = std::result::Result<T, E>;
 /// Result type of a pure HTTP request.
 pub type HttpResult<T> = std::result::Result<T, HttpError>;
 
+/// An error response from a Matrix API call, using a client API specific
+/// representation if the endpoint is from that.
+#[derive(Error, Debug)]
+pub enum RumaApiError {
+    #[error("{0}")]
```

Comment:
Isn't this the same as `#[error(transparent)]`?

Reply:
I wasn't aware of that :)

Reply:
Okay so it's a little bit different in that it also delegates the `source` method. Not sure whether we want that. It prevents using source to downcast to these Ruma types, but I don't see why anybody would do that in the first place 🤷🏼